wallet/src/Pos/Wallet/Web/Methods/History.hs

```diff
@@ -122,20 +129,39 @@ addHistoryTx
     :: MonadWalletWebMode m
     => CId Wal
     -> TxHistoryEntry
+    -> m ()
+addHistoryTx cWalId = addHistoryTxs cWalId . one
+
+-- This functions is helper to do @addHistoryTx@ for
+-- all txs from mempool as one Acidic transaction.
+addHistoryTxs
+    :: MonadWalletWebMode m
+    => CId Wal
+    -> [TxHistoryEntry]
+    -> m ()
+addHistoryTxs cWalId historyEntries = do
+    metas <- mapM toMeta historyEntries
+    let cIds = map (encodeCType . _thTxId) historyEntries
+    addOnlyNewTxMetas cWalId (zip cIds metas)
```

Comment:
Here acid state will create event log of size `O(|mempool|)`, and it happens on each request to `getHistory`.
But at present mempool is mostly empty for wallet users, so this is minor